### PR TITLE
[core] {TRUST/PET/MOB} shield block functionality

### PR DESCRIPTION
**Pull request by GodModeXI**
…ect exception / add shield skill calculation / adjust reprisal effect to generate spike damage

@Schaedelschaden supported with printf audit and debugging 
 reopened from #3037 

**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?
- This PR corrects issue #2926 
- Adds a shield base rate to trusts bypassing shield size calculation (since trusts have no gear)
- Adds in Shield skill to assist in stat calculations.
- Adds subeffect functionality for Reprisal for trusts that bypasses the shield check logic
- Adds a power calculation to the Reprisal effect lua to not return 0 damage. 
- Audit Data
blockrates
    - Block Rates
        - 50 = guaranteed 5% + 45% base applied in Core
        - 97 = base 50 + Reprisal calculations
        - 80 = base 50 + Palisade calculations

reprisal no dmg
    - Reprisal : No spike damage

reprisal with damage
    - Reprisal : with effect power added

## Steps to test these changes

1. Load in core change files and reprisal effect lua
2. Clean and rebuild server to apply changes.
3. load a PLD trust with palisade / reprisal logic applied.
4. add a printf() into battleutils.cpp before the return to show block rates.

### CODE REVIEW

File: src/map/utils/battleutils.cpp [hunk 1]
```diff
@@ -1802,27 +1803,49 @@ namespace battleutils
...
+                    }
+                    else
+                    {
+                        return base + skillModifier;
+                    }
+                }
+                else
+                {
+                    // Check for Reprisal and adjust skill and block rate bonus multiplier
+                    if (PDefender->StatusEffectContainer->HasStatusEffect(EFFECT_REPRISAL))
```
Comment (WinterSolstice8) [inline src/map/utils/battleutils.cpp:1830]: We need a skill modifier check --
      
```
            // TODO: check trust type for ilvl > 99 when implemented
            blockSkill = GetMaxSkill(SKILLTYPE::SKILL_SHIELD, PDefender->GetMJob(), PDefender->GetMLevel() > 99 ? 99 : PDefender->GetMLevel());
```
 need the multiplier for block skill again from reprisal + the other block skill modifiers in from type PC

**Review by WinterSolstice8 — CHANGES_REQUESTED** [review]

File: src/map/utils/battleutils.cpp (hunk 1 above)
Comment (GodModeXI) [reply src/map/utils/battleutils.cpp:1830]: That's the bit I was missing before Teo kept trying to tell me to put in but I misunderstood him. 😭

Comment (GodModeXI) [reply src/map/utils/battleutils.cpp:1830]: > We need a skill modifier check --
> 
>       
> 
> ```
> 
>             // TODO: check trust type for ilvl > 99 when implemented
> 
>             blockSkill = GetMaxSkill(SKILLTYPE::SKILL_SHIELD, PDefender->GetMJob(), PDefender->GetMLevel() > 99 ? 99 : PDefender->GetMLevel());
> 
> ```
> 
>  need the multiplier for block skill again from reprisal + the other block skill modifiers in from type PC
> 
> 

- Block flat values prior to skill check
    - 30 (flat (mod))
    - 45 (+15% reprisal)
    - 60 (+30% palisade)
image

- Block with max skill check added
    - 35 (flat (mod) +5%)
    - 74 (+39% Reprisal)
    - 65 (+30% palisade)
image

**Review by WinterSolstice8 — APPROVED** [review]

**Review by TeoTwawki — APPROVED** [review]
lgtm now

**Pull request merged**